NitroxPatcher/Patches/Dynamic/FreezeTime_Begin_Patch.cs

```diff
@@ -1,28 +1,28 @@
-﻿using System.Reflection;
+using System.Collections.Generic;
+using System.Reflection;
 using HarmonyLib;
 using NitroxModel.Helper;
 using UWE;
 
 /// <summary>
 /// Because we're in multiplayer mode, we generally don't want the game to freeze
 /// </summary>
-namespace NitroxPatcher.Patches.Dynamic
+namespace NitroxPatcher.Patches.Dynamic;
+
+public class FreezeTime_Begin_Patch : NitroxPatch, IDynamicPatch
 {
-    public class FreezeTime_Begin_Patch : NitroxPatch, IDynamicPatch
-    {
-        private static readonly MethodInfo TARGET_METHOD = Reflect.Method(() => FreezeTime.Begin(default(FreezeTime.Id)));
+    private static readonly MethodInfo TARGET_METHOD = Reflect.Method(() => FreezeTime.Begin(default(FreezeTime.Id)));
+
+    private static HashSet<FreezeTime.Id> allowedFreezeIds = new() { FreezeTime.Id.Quit, FreezeTime.Id.WaitScreen };
 
-        // We don't want to prevent from freezing the game if the opened modal wants to freeze the game
-        public static bool Prefix(string userId)
-        {
-            // If we ask to freeze from a Nitrox modal, userId will be like this: NitroxServerStoppedModalFreeze
-            return userId.Contains("Nitrox") && userId.EndsWith("Freeze");
-        }
+    // We don't want to prevent from freezing the game if the opened modal wants to freeze the game
+    public static bool Prefix(FreezeTime.Id id)
+    {
+        return allowedFreezeIds.Contains(id);
```

Comment:
It's likely we will never have update on this enum, maybe we can just do some ==

Reply:
I think this is mostly stylistic as there isn't any performance implications

Reply:
Ran a benchmark. First two rows return true, last return false. Performance difference is relatively huge but let's say anything 0.001ms (1000ns) or lower should be fine.

The hash set adds extra context with its variable name. It's very minor. But if I had to decide I agree with Clement here because the comments are already explaining the code well.

|      Method |         id |      Mean |     Error |    StdDev | Ratio | Allocated | Alloc Ratio |
|------------ |----------- |----------:|----------:|----------:|------:|----------:|------------:|
|     Hashset | WaitScreen | 7.4625 ns | 0.0170 ns | 0.0159 ns | 1.000 |         - |          NA |
| Conditional | WaitScreen | 0.0146 ns | 0.0036 ns | 0.0034 ns | 0.002 |         - |          NA |
|             |            |           |           |           |       |           |             |
|     Hashset |  Something | 5.3859 ns | 0.0153 ns | 0.0136 ns | 1.000 |         - |          NA |
| Conditional |  Something | 0.0372 ns | 0.0031 ns | 0.0026 ns | 0.007 |         - |          NA |

Reply:
Sorry I left the comment ambiguous. I think it is obvious that a hashset has poorer performance characteristics over a hard coded conditional.  The context is that when someone is playing then FreezeTime.Begin is invoked when dialogs like save/quit are opened.  The performance implication is negligible.  You probably get more overhead from the harmony method proxy.